src/RobotInterface/include/BipedalLocomotion/RobotInterface/ISensorBridge.h

```diff
@@ -0,0 +1,349 @@
+     * @param[out] ftMeasurement FT measurements of size 3 containing tau_x tau_y and fz
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getThreeAxisForceTorqueMeasurement(const std::string& ftName,
+                                                    Eigen::Ref<Eigen::Vector3d> ftMeasurement,
+                                                    double* receiveTimeInSeconds = nullptr) = 0;
+
+   /**
+     * Get 6D end effector wrenches in N and Nm for forces and torques respectively
+     * @param[in] cartesianWrenchName name of the end effector wrench
+     * @param[out] cartesianWrenchMeasurement end effector wrench measurement of size 6
+     * @param[out] rreceiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getCartesianWrench(const std::string& cartesianWrenchName,
+                                    Eigen::Ref<Vector6d> cartesianWrenchMeasurement,
+                                    double* receiveTimeInSeconds = nullptr) = 0;
+
+    /**
+     * Get color image from the camera
+     * @param[in] camName name of the camera
+     * @param[out] colorImg image as Eigen matrix object
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getColorImage(const std::string& camName,
+                               Eigen::Ref<Eigen::MatrixXd> colorImg,
+                               double* receiveTimeInSeconds = nullptr) = 0;
+
+    /**
+     * Get depth image
+     * @param[in] camName name of the gyroscope
+     * @param[out] depthImg depth image as a Eigen matrix object
+     * @param[out] receiveTimeInSeconds time at which the measurement was received
+     * @return true/false in case of success/failure
+     */
+    virtual bool getDepthImage(const std::string& camName,
+                               Eigen::Ref<Eigen::MatrixXd> depthImg,
+                               double* receiveTimeInSeconds = nullptr) = 0;
+
+    /**
+     * Destructor
+     */
+    virtual ~ISensorBridge() = default;
+
+protected:
+    /**
+     * Helper method to maintain SensorBridgeOptions struct by populating it from the configuration parameters
+     * @note the user may choose to use/not use this method depending on their requirements for the implementation
+     * if the user chooses to not use the method, the implementation must simply contain "return true;"
+     *
+     * @param[in] handler  Parameters handler
+     * @param[in] sensorBridgeOptions SensorBridgeOptions to hold the bridge options for streaming sensor measurements
+     */
+    virtual bool populateSensorBridgOptionsFromConfig(std::weak_ptr<IParametersHandler> handler,
+                                                      SensorBridgeOptions& sensorBridgeOptions) = 0;
```

Comment:
If it is not compulsory, simply return true directly in the header. Otherwise, you can already provide a basic implementation that can be overridden by the user, or even not used since it is protected.

Reply:
I refrained myself from doing that to keep this class abstract. If you think it is not necessarily important to remain as a pure virtual class, I shall add the implementation that simply returns true in the header.

Reply:
Yes, in C++ a class is abstract if at least a method is purely virtual, so no big deal.

Reply:
> A class is made abstract by declaring at least one of its functions as a pure virtual function.

Ah, yes! I was thinking conversely on this.

 I will make the suggested changes then (returning true would be easy :D).

Thank you for this suggestion.

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/87/commits/979733e1357ffed832c1533a6c6c464adcdae6b8